snuba/datasets/cdc/groupassignee.py

```diff
@@ -68,10 +69,8 @@ def __init__(self) -> None:
             version_column="offset",
         )
 
-        dataset_schemas = DatasetSchemas(read_schema=schema, write_schema=schema,)
-
-        super().__init__(
-            dataset_schemas=dataset_schemas,
+        storage = WritableTableStorage(
+            schemas=StorageSchemas(read_schema=schema, write_schema=schema),
```

Comment:
Reading this again in context: are the multiple storages here just as a transitional thing to minimize this change? It seems like a storage should only have one schema, especially given that the storage selection is now an abstraction on its own.

EDIT: OK, came back across this comment, which somewhat answers my question: https://github.com/getsentry/snuba/pull/828/files#r391807959 Personally, I think it would make more sense for a storage to only have one schema (so any materialized views would also be defined as separate storages) and the dependency graph defined at the storage level instead of the schema level.

Reply:
> are the multiple storages here just as a transitional thing to minimize this change?

Yes, any changes to the structure or responsibility of the StorageSchemas class (former DatasetStorages) would require an extensive refactoring of the DDL, bootstrap and migrations code.

>  It seems like a storage should only have one schema

That's my idea for most storages. A dataset like this would only declare one schema for the only storage it uses. It has to pass it twice now because we are in the process of moving the distinction between read schema and write schema from StorageSchemas to the dataset itself by introducing Storages.
The only case in which, today, we would need multiple schemas per storages are matviews, and this is not because they have a read schema and a write schema (those are already two separate storages in this PR). The reason is that they have an intermediate third schema that clickhouse uses to run the materialization query. 
So read schema -> read storage, write schema -> write storage. This is easy. Whether we can make the intermediate schema a storage on its own, I am not sure, it was one of the possible directions but I did not try to see if it works yet.

I could introduce the third storage in this PR, but I still cannot get rid of StorageSchemas otherwise the issue is the same of the one at the beginning of this comment.